### PR TITLE
Regularly try to reconnect disconnected devices (#3299)

### PR DESCRIPTION
**Pull request by freddidierRTE**
Signed-off-by: freddidierRTE <frederic-f.didier@rte-france.com>

- In release note :
  -  In chapter :  Features 
  -  Text : Regularly try to reconnect disconnected external devices (#3299)

### CODE REVIEW

File: src/test/resources/external-devices/createDevice.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+url=$2 
+if [ -z $url ] 
+then
+	url="http://localhost"
+fi
+if [ -z $1 ]
+then
+    echo "Usage : createDevice file_name opfab_url"
```
Comment (vlo-rte) [inline src/test/resources/external-devices/createDevice.sh:22]: createDevice => createDevice.sh

File: src/test/resources/external-devices/deleteDevice.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+url=$2
+if [ -z $url ] 
+then
+	url="http://localhost"
+fi
+if [ -z $1 ]
+then
+    echo "Usage : deleteDevice device_id opfab_url"
```
Comment (vlo-rte) [inline src/test/resources/external-devices/deleteDevice.sh:22]: deleteDevice => deleteDevice.sh

File: src/test/resources/external-devices/disconnectDevice.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+url=$2 
+if [ -z $url ] 
+then
+	url="http://localhost"
+fi
+if [ -z $1 ]
+then
+    echo "Usage : connectDevice.sh deviceId opfab_url"
```
Comment (vlo-rte) [inline src/test/resources/external-devices/disconnectDevice.sh:22]: connectDevice.sh => disconnectDevice.sh

**Review by vlo-rte — CHANGES_REQUESTED** [review]

File: services/external-devices/src/main/java/org/opfab/externaldevices/services/DevicesService.java [hunk 4]
```diff
@@ -39,144 +28,213 @@
...
+                    resolvedConfiguration.getDeviceConfiguration());
+            if (!externalDeviceDriver.isConnected()) {
+                log.info("External device {} was not connected. Connecting before send.",
+                        deviceId);
+                connectDevice(resolvedConfiguration.getDeviceConfiguration());
+            }
+            externalDeviceDriver.send(signalId);
+        } else
+            throw new ExternalDeviceAvailableException(
+                    String.format("Could not send signal to device %s because it is disabled", signalId));
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply services/external-devices/src/main/java/org/opfab/externaldevices/services/DevicesService.java:213]: yes

**Review by vlo-rte — APPROVED** [review]

**Pull request merged**